tests/run_pretrained_models.py

```diff
@@ -246,9 +246,9 @@ def run_test(self, name, backend="caffe2", debug=False, onnx_file=None, opset=No
             for k in inputs.keys():  # pylint: disable=consider-iterating-dictionary
                 t = sess.graph.get_tensor_by_name(k)
                 dtype = tf.as_dtype(t.dtype).name
-                if type != "float32":
-                    v = inputs[k]
-                    inputs[k] = v.astype(dtype)
+                v = inputs[k]
+                if dtype != v.dtype:
```

Comment:
inputs is provided in yaml, right? If yes, shall we add a warning if yaml giving a type not matching tf type? this should be a error for yaml config I feel.